libs/barista-components/table/src/table.ts

```diff
@@ -99,7 +102,7 @@ export class DtTable<T> extends _DtTableBase<T> implements OnDestroy {
   private _loading: boolean;
   private _destroy$ = new Subject<void>();
 
-  private _showExportButton: boolean = false; //Revert to opt-in instead of opt-out per request
+  private _showExportButton: boolean | 'visible' | 'table' = false; //Display button for Visible, Table, both (true), or neither (false)
```

Comment:
What exactly is the difference between true and `'visible'`? That type seems a bit complicated, can you please elaborate why this extension was necessary?

Reply:
`visible` only gives the option for the Visible data versus both visible and underlying table data. Mohamed had a usecase where the table data includes many more columns than what’s visible. This would be an alternative to a very long exclude list. I’ll update docs to indicate this is an option, but that devs should prefer `true`.